cmd/kube-apiserver/app/server.go

```diff
@@ -552,9 +554,49 @@ func BuildAuthenticator(s *options.ServerRunOptions, storageFactory serverstorag
 	return authenticatorConfig.New()
 }
 
+// buildAuthorizationWebhookDialer constructs a custom dialer function for the authorization webhook
+// that knows how to resolve references to Kubernetes services (of the form "servicename.namespace.svc")
+// to the IP address of the endpoint that currently implements it.
+func buildAuthorizationWebhookDialer(
+	serviceResolver aggregatorapiserver.ServiceResolver,
+	proxyTransport *http.Transport,
+) func(network, address string) (net.Conn, error) {
+	// Following sample code from issue #54163 by liggitt@redhat.com.
+	baseDialer := net.Dial
+	if proxyTransport != nil && proxyTransport.Dial != nil {
+		baseDialer = proxyTransport.Dial
+	}
+
+	return func(network, address string) (net.Conn, error) {
+		host, _, err := net.SplitHostPort(address)
+		if err != nil {
+			glog.Warning("while splitting hostport: %v", err)
```

Comment:
for debugging, I'm assuming... I'd recommend just passing through and letting the underlying dialer return an error about a malformed address

Reply:
Changed to a verbose log message. If you don't mind, I'd leave a verbose log here.  While yes, passing through will eventually surface an error message, it gets very difficult to figure out the error message source if you are not very familiar with the code path.  This log will reveal at least part of it.  This has, at least to me, been useful numerous times in debugging.